services/team.service.js

```diff
@@ -46,6 +49,91 @@ function findById(id) {
     return Team.findById(query, logger.queryCallbackFactory(TAG, "team", query));
 }
 
+/**
+ * @function findByName
+ * @param {String} name
+ * @return {DocumentQuery} The document query will either resolve to a team or null.
+ * @description Finds a team by its team name.
+ */
+function findByName(name) {
+    const TAG = `[Team Services # findByName]:`;
+
+    const query = {
+        name: name
+    };
+
+    return Team.findOne(query, logger.queryCallbackFactory(TAG, "team", query));
+}
+
+/**
+ * @function removeMember
+ * @param {ObjectId} teamId
+ * @param {ObjectId} hackerId
+ * @return {DocumentQuery} The document query will resolve to the number of objects removed, or null.
+ * @description Removes the hacker specified by hackerId from a team specified by teamId.
+ */
+function removeMember(teamId, hackerId) {
+    const TAG = `[Team Services # removeMember]:`;
+
+    return Team.update({
+        _id: teamId
+    }, {
+        $pull: {
+            members: [hackerId]
+        }
+    });
+}
+
+/**
+ * @function addMember
+ * @param {ObjectId} teamId
+ * @param {ObjectId} hackerId
+ * @return {DocumentQuery} Query evaluates to object that details the number of modified documents, or null.
+ * @description Add the hacker specified by hackerId to the team specified by teamId
+ */
+function addMember(teamId, hackerId) {
+    const TAG = `[Team Services # addMember]:`;
+
+    return Team.update({
+        _id: teamId
+    }, {
+        $push: {
+            members: [hackerId]
+        }
+    });
+}
+
+/**
+ * @function removeTeam
+ * @param {ObjectId} teamId
+ * @return {DocumentQuery} The document query will resolve to the number of objects removed, or null.
+ * @description Delete the team specified by teamId. 
+ */
+function removeTeam(teamId) {
+    const TAG = `[Team Services # removeTeam]`;
+
+    return Team.deleteOne({
```

Comment:
This should delete the team from all of the hacker objects listed in it as well.

Reply:
Right, same with the removeMember and addMembver stuff. They should also modify the hackers.